core/dbt/deps/local.py

```diff
@@ -51,19 +51,15 @@ def install(self, project, renderer):
         src_path = self.resolve_path(project)
         dest_path = self.get_installation_path(project, renderer)
 
-        can_create_symlink = system.supports_symlinks()
-
         if system.path_exists(dest_path):
             if not system.path_is_symlink(dest_path):
                 system.rmdir(dest_path)
             else:
                 system.remove_file(dest_path)
-
-        if can_create_symlink:
```

Comment:
`if can_create_symlink` check shouldn't be removed. It helps for the cases where `symlink` function doesn't exist on `os`, in which case `AttributeError` would be thrown.

`except OSError` should be added as an edge case like ours, where `os.symlink` function exists but it still fails.

Something like:
```
symlink_supported = False

if can_create_symlink:
    try:
        fire_event(DepsCreatingLocalSymlink())
        system.make_symlink(src_path, dest_path)
    except OSError:
        pass
    else:
        symlink_supported = True
if not symlink_supported:
    fire_event(DepsSymlinkNotAvailable())
    shutil.copytree(src_path, dest_path)
```

Reply:
I do not think we need it, `make_symlink`  calling it. The functionality remains the same. Check make_symlink method.

Reply:
`make_symlink` method will throw `dbt.exceptions.SymbolicLinkError`, in which case `except OSError` block here won't handle it.  
The behaviour is changed for the case where `symlink` function attribute isn't present on `os`

Reply:
yes, I think we should use `dbt.exceptions.SymbolicLinkError,` and we should also catch the SymbolicLinkError. 

Also, I am curious about which OS does not have symlinks function in python `lib(os)` package, as mentioned in my Issue description.
As per the dbt, we have a list of supported operating systems and condition on python3.7+. https://docs.getdbt.com/faqs/core/install-pip-os-prereqs.md

If my understanding is true, we may not have any scenarios where  `lib(os)` package will not have a `symlink` function and we may not need to handle the SymbolicLinkError exception. 

Waiting for the core reviewers to help us here.